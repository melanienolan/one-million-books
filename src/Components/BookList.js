import React from 'react';
import PropTypes from 'prop-types';

const BookList = props => {
  const renderList = () => {
    return props.books.map(book => {
      return (
        <li className="book--item" key={book.id}>
          <p className="book--item-text">
            <span className="book--item-field">Title:</span> {book.title}
          </p>
          <p className="book--item-text">
            <span className="book--item-field">Author:</span> {book.authorName}
          </p>
          <p className="book--item-text">
            <span className="book--item-field">Author Gender:</span> {book.authorGender}
          </p>
          <p className="book--item-text">
            <span className="book--item-field">Genre:</span> {book.genre}
          </p>
          <p className="book--item-text">
            <span className="book--item-field">Date published:</span> {book.published}
          </p>
        </li>
      );
    });
  };
  return (
    <ul className="book--list">
      {renderList()}
    </ul>
  );
};

BookList.propTypes = {
  books: PropTypes.array
};

export default BookList;
