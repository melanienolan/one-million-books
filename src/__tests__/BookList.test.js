import React from 'react';
import BookList from '../Components/BookList';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import db from '../mockDB';

Enzyme.configure({ adapter: new Adapter() });

it('renders book list component with books without crashing', () => {
  const { books } = db;
  Enzyme.shallow(<BookList books={books} />);
});
it('renders book list component with no books without crashing', () => {
  const books = [];
  Enzyme.shallow(<BookList books={books} />);
});

it('renders book list error message with no books', () => {
  const books = [];
  const bookListComponent = Enzyme.shallow(<BookList books={books} />);
  expect(toJson(bookListComponent)).toMatchSnapshot();
});
