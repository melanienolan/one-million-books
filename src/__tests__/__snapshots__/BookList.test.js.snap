// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders book list error message with no books 1`] = `
<div
  className="table--empty"
>
  <div
    className="table--error"
  >
    No books exist with those filters
  </div>
  <p
    className="table--number"
  >
    Number of books showing: 
    0
  </p>
</div>
`;
